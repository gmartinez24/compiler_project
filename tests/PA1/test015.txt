// Andrew Jack Roehr
// Testing negative number/decrement ambiguity
// Test further comments
// These both tripped up my parser a bit

main

int test_neg_int = -12;
float test_neg_float = -3.0;
test_neg_int--;

/* asdf
  */ // qwerty
test_neg_float--;
/* end test */
