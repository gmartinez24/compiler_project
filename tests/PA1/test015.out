MAIN
INT
IDENT	test_neg_int
ASSIGN
INT_VAL	-12
SEMICOLON
FLOAT
IDENT	test_neg_float
ASSIGN
FLOAT_VAL	-3.0
SEMICOLON
IDENT	test_neg_int
UNI_DEC
SEMICOLON
IDENT	test_neg_float
UNI_DEC
SEMICOLON
EOF
